### PR TITLE
Add Codespaces JS project workshop

### PR DESCRIPTION
**Pull request by saragibby**
## Use this template to add a workshop to this repo. Add the following items:

Adds a workshop offering for [JavaScript Portfolio Site with GitHub Codespaces](https://github.com/microsoft/codespaces-project-template-js) template repo project.

- [X] Your workshop design document, filled in with some details about your proposed workshop including its title and the three stages in the designer delineated, using the `workshop-designer.md` file in the template.
- [X] A `README.md` file that describes the steps of the workshop. 
- [X] PowerPoint Slides for the workshop.
- [X] Code quality check: no authentication keys should be included in solution or sample code

A reviewer will be assigned to the workshop and will work with you to shape it for production. Thank you for your contribution!

### CODE REVIEW

File: short/codespaces-project-template-js/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+You can also configure your development container to run a specific runtime and have it boot up with your favorite extensions.
+
+## What you will learn and requirements for this workshop
+
+| **Goal**              | Description                                    |
+| ----------------------------- | --------------------------------------------------------------------- |
+| **What will you learn**       | customizing Codespaces, web development, and web application deployment              |
+| **What you'll need**          | <ul><li>[GitHub account](https://github.com)</li><li>access to [Codespaces](https://github.com/features/codespaces)</li><li> optional [Azure for Students](https://azure.microsoft.com/en-us/free/students/) or [Azure](https://azure.microsoft.com/en-us/free/search/) account for deployment</li></ul> |
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/README.md:22]: The azure links need to have WT links and get the localization (`en-us`) removed

File: short/codespaces-project-template-js/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+This template [JavaScript Portfolio Site with GitHub Codespaces](https://github.com/microsoft/codespaces-project-template-js) includes all the resources you **will need in order to demonstrate the full presentation to your audience**.
+
+Within this workshop, there is [3 points of customization](https://github.com/microsoft/codespaces-project-template-js#-customize-your-site-in-3-steps) included in the project template repository to be included. This will allow each attendee to customize their own site.
+
+Beyond 3 points of customization, there are [4 challenge tasks](https://github.com/microsoft/codespaces-project-template-js#-challenges). These allow you to customize this workshop to your audience and time you;d like to spend.
+
+Follow the [getting started instructions](https://github.com/microsoft/codespaces-project-template-js#-getting-started) to get your workshop attendees started with their poject launched within [Codespaces](https://github.com/features/codespaces). 
+
+<br />
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/README.md:38]: are these tags needed?

File: short/codespaces-project-template-js/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## 🏃 Deploy your web application
+
+Project includes the setup needed for you to deploy **free** to both <a href="https://azure.microsoft.com/en-in/products/app-service/static/" target="_blank">Azure Static Web Apps</a> and <a href="https://pages.github.com/" target="_blank">GitHub Pages</a>.
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/README.md:43]: This Azure link will also need updating

File: short/codespaces-project-template-js/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+### Azure Static Web Apps
+
+<a href="https://azure.microsoft.com/en-in/products/app-service/static/" target="_blank">Azure Static Web Apps</a> is Microsoft's hosting solution for static sites (or sites that are rendered in the user's browser, not on a server) through Azure. This service provides additional opportunities to expand your site through Azure Functions, authentication, staging versions and more.
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/README.md:47]: Please update this link as well to remove localization and to use a WT link

File: short/codespaces-project-template-js/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+You'll need both Azure and GitHub accounts to deploy your web application. If you don't yet have an Azure account you can create it from within during the deploy process, or from below links:
+
+* [Create a (no Credit Card required) Azure For Students account](https://azure.microsoft.com/free/students/)
+* [Create a new Azure account](https://azure.microsoft.com/en-US/)
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/README.md:52]: Please update these links as well to remove localization and to use a WT link

File: short/codespaces-project-template-js/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+1. Commit and push those changes to `package.json` to your GitHub remote repo.
+1. Open a new terminal from the menu or by pressing `ctrl` + `shift` + ` (or open top left menu, select "Terminal" and "New Terminal")
+1. Within the terminal window run `npm run deploy`. This will first run the pre-deploy script to build the project, followed by the deploy script that will push those bundled files to a new branch on your repo (gh-pages) that will be used for you GitHub Pages site.
+1. When completed, within your repo, go to Settings and Pages. There you will see that a page has been set up to for you against the gh-pages branch, and you should see the URL (that should match the “homepage” value you set in package.json)
+
+![Azure Static Web App deploy](./images/github-pages-deploy.gif "Azure Static Web App deploy")
+
+<br />
+
+## 📚 Resources
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/README.md:87]: Can you add the https://aka.ms/CodespacesVideoTutorial link here in the resources?

**Review by alfredodeza — CHANGES_REQUESTED** [review]

File: short/codespaces-project-template-js/README.md (hunk 2 above)
Comment (saragibby) [reply short/codespaces-project-template-js/README.md:38]: They are, I am referring to specific sections in that readme here.

File: short/codespaces-project-template-js/README.md (hunk 6 above)
Comment (saragibby) [reply short/codespaces-project-template-js/README.md:87]: added

File: short/codespaces-project-template-js/README.md (hunk 1 above)
Comment (saragibby) [reply short/codespaces-project-template-js/README.md]: Thanks or catching this localized route! All have been removed.

File: short/codespaces-project-template-js/workshop-designer.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+
+# JavaScript Portfolio Site with GitHub Codespaces
+
+## Workshop Source
+
+https://docs.microsoft.com/learn/modules/publish-app-service-static-web-app-api/
```
Comment (alfredodeza) [inline short/codespaces-project-template-js/workshop-designer.md:6]: This one is missing a WT link

**Review by alfredodeza — APPROVED** [review]
This looks good to go Sarah, just one more link to update. Feel free to merge once that last change is done. Thank you! 🚀

**Pull request merged**